app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationScreen.kt

```diff
@@ -288,7 +293,16 @@ fun ConversationScreen(
         requestMentions = messageComposerViewModel::searchMembersToMention,
         onClearMentionSearchResult = messageComposerViewModel::clearMentionSearchResult,
         conversationScreenState = conversationScreenState,
-        messageComposerStateHolder = messageComposerStateHolder
+        messageComposerStateHolder = messageComposerStateHolder,
+        onLinkClick = { link ->
+            with(messageComposerViewModel) {
+                if (isLinkValid(link)) {
+                    uriHandler.openUri(link)
+                } else {
+                    invalidLinkDialogState = InvalidLinkDialogState.Visible
+                }
+            }
```

Comment:
not really sure, but couldn't we have this inside a function in the ViewModel instead?

Reply:
oooh I missed that!